src/app/shared/search/search-filters/search-filter/search-facet-filter/search-facet-filter.component.ts

```diff
@@ -324,6 +275,67 @@ export class SearchFacetFilterComponent implements OnInit, OnDestroy {
     return getFacetValueForType(facet, this.filterConfig);
   }
 
+  protected retrieveFilterValues(useCachedVersionIfAvailable = true) {
+    const facetValues$ = observableCombineLatest([this.searchOptions$, this.currentPage]).pipe(
+      map(([options, page]) => {
+        return { options, page };
+      }),
+      switchMap(({ options, page }) => {
+        return this.searchService.getFacetValuesFor(this.filterConfig, page, options, null, useCachedVersionIfAvailable)
+          .pipe(
+            getFirstSucceededRemoteData(),
+            map((results) => {
+                return {
+                  values: observableOf(results),
+                  page: page
+                };
+              }
+            )
+          );
+      })
+    );
+
+    let filterValues = [];
+    this.subs.push(facetValues$.subscribe((facetOutcome) => {
+      const newValues$ = facetOutcome.values;
+
+      if (this.collapseNextUpdate) {
+        this.showFirstPageOnly();
+        facetOutcome.page = 1;
+        this.collapseNextUpdate = false;
+      }
+      if (facetOutcome.page === 1) {
+        filterValues = [];
+      }
+
+      filterValues = [...filterValues, newValues$];
+
+      this.subs.push(this.rdbs.aggregate(filterValues).pipe(
+        tap((rd: RemoteData<PaginatedList<FacetValue>[]>) => {
+          this.selectedValues$ = this.filterService.getSelectedValuesForFilter(this.filterConfig).pipe(
+            map((selectedValues) => {
+              return selectedValues.map((value: string) => {
+                const fValue = [].concat(...rd.payload.map((page) => page.page)).find((facetValue: FacetValue) => this.getFacetValue(facetValue) === value);
+                if (hasValue(fValue)) {
+                  return fValue;
+                }
+                const filterValue = stripOperatorFromFilterValue(value);
+                return Object.assign(new FacetValue(), { label: filterValue, value: filterValue });
+              });
+            })
+          );
+        })
+      ).subscribe((rd: RemoteData<PaginatedList<FacetValue>[]>) => {
+        this.animationState = 'ready';
+        this.filterValues$.next(rd);
+
+      }));
+      this.subs.push(newValues$.pipe(take(1)).subscribe((rd) => {
```

Comment:
This `subscribe` shouldn't be here if this method is called from another `subscribe`

Reply:
done